andela/bp-esa-backend
# #162548574 Refactor slackIntegrations

#### What does this PR do?
Refactor slack integrations to eliminate redundant codes

#### Description of Task to be completed?
- `catch` statements which return error should be removed so that the error would be thrown when the function is called
- `catch` blocks which just throws an exception should be removed
- redundant `createChannel` function which adds cognitive complexity to the implementation should be removed and implementation refactored
- redundant `channelExists` function which is useless since slack api throws an error when the channel to be created already exists, should be removed
- redundant `returnValidChannels` function which is based on redundant `channelExists` function should be removed, being no longer needed

#### How should this be manually tested?
1. Call the various functions in the module and log their output
2. On terminal, run command `npx babel-node erver/modules/slack/slackIntegration.js`
3. Watch the terminal to confirm the function works as expected

#### Any background context you want to provide?
- This would make it easier to catch the actual error in run time

#### What are the relevant pivotal tracker stories?
[#162548574](https://www.pivotaltracker.com/story/show/162548574)

#### Postman Documentation
N/A
